palace/fem/fespace.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef PALACE_FEM_FESPACE_HPP
+#define PALACE_FEM_FESPACE_HPP
+
+#include <mfem.hpp>
+
+namespace palace
+{
+
+//
+// Wrapper for MFEM's ParFiniteElementSpace class, where the finite element space object
+// is constructed with a unique ID associated with it. This is useful for defining equality
+// operations between spaces (either different spaces on the same mesh, or the same space
+// type on different meshes).
+//
+class FiniteElementSpace : public mfem::ParFiniteElementSpace
+{
+private:
+  static std::size_t global_id;
+  mutable std::size_t id;
```

Comment:
What's the distinction between the `global_id` and `id`?

Reply:
`global_id` is `static`, there is only one instance of it across all `FiniteElementSpace` objects. It's used as a global counter. `id` is local to each instance of a `FiniteElementSpace` and is guaranteed to not be the same as any other `FiniteElementSpace`.